services/user-service/api/handler/login_test.go

```diff
@@ -0,0 +1,178 @@
+package handler
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/hingew/hsfl-master-ai-cloud-engineering/lib/gomockhelpers"
+	mocks "github.com/hingew/hsfl-master-ai-cloud-engineering/user-service/_mocks"
+	"github.com/hingew/hsfl-master-ai-cloud-engineering/user-service/user/model"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+)
+
+func TestLoginHandler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	userRepository := mocks.NewMockRepository(ctrl)
+	hasher := mocks.NewMockHasher(ctrl)
+	tokenGenerator := mocks.NewMockTokenGenerator(ctrl)
+	handler := NewLoginHandler(userRepository, hasher, tokenGenerator)
+
+	t.Run("should return 405 METHOD NOT ALLOWED if method is not POST", func(t *testing.T) {
+		// given
+		w := httptest.NewRecorder()
+		r := httptest.NewRequest("GET", "/api/v1/auth/login", nil)
+
+		// when
+		handler.ServeHTTP(w, r)
+
+		// then
+		assert.Equal(t, http.StatusMethodNotAllowed, w.Code)
+	})
+
+	t.Run("should return 400 BAD REQUEST if payload is not json", func(t *testing.T) {
+		tests := []io.Reader{
+			nil,
+			strings.NewReader(`{"invalid json`),
+		}
+
+		for _, test := range tests {
+			// given
+			w := httptest.NewRecorder()
+			r := httptest.NewRequest("POST", "/api/v1/auth/login", test)
+
+			// when
+			handler.ServeHTTP(w, r)
+
+			// then
+			assert.Equal(t, http.StatusBadRequest, w.Code)
+		}
+	})
+
+	t.Run("should return 400 BAD REQUEST if payload is incomplete", func(t *testing.T) {
+		tests := []io.Reader{
+			strings.NewReader(`{"email":"test@test.com"}`),
+			strings.NewReader(`{"password":"test"}`),
+		}
+
+		for _, test := range tests {
+			// given
+			w := httptest.NewRecorder()
+			r := httptest.NewRequest("POST", "/api/v1/auth/login", test)
```

Comment:
Why do we have a v1 prefix here?

Reply:
hab mal die Versionierung entfernt

Reply:
:+1: Ich denke für erstmal sollten wir auf die Versionierung der API verzichten, da wir im moment nur eine Version haben.